tensorflow/compiler/jit/xla_compilation_cache.cc

```diff
@@ -187,21 +189,30 @@ Status XlaCompilationCache::BuildExecutable(
 
 Status XlaCompilationCache::Compile(
     const XlaCompiler::Options& options, const NameAttrList& function,
-    absl::Span<const XlaCompiler::Argument> args,
+    const std::vector<XlaCompiler::Argument>& args,
     const XlaCompiler::CompileOptions& compile_options,
     CompileMode compile_mode,
     const XlaCompiler::CompilationResult** out_compilation_result,
     xla::LocalExecutable** out_executable) {
-  absl::optional<int64> compile_threshold;
-  if (compile_mode == CompileMode::kLazy) {
-    compile_threshold = kDefaultCompilationThreshold;
-  }
-  auto compile_fn = [&](XlaCompiler* compiler,
+  // Explicitly capture all required data by value for async compilation
+  struct AsyncCompileData {
+    const XlaCompiler::CompileOptions& compile_options;
+    const NameAttrList& function;
```

Comment:
Please add a comment here describing why references are OK (and won't cause a use-after-free).

Reply:
I can't , because the code is incorrect. Good catch.

Reply:
So I see no way do do this w/o copying items twice (populate the struct, copy by value).
I don't want to do that (function is a map, and copying twice is bad)

Reply:
I'm reverting back to my original version, and will add a comment to pay  attention.
What you want looks safer, but comes at a cost of having to copy non-POD times twice